src/invoice/mod.rs

```diff
@@ -96,9 +97,24 @@ impl Invoice {
         version_compare(self.bindle.id.version(), requirement)
     }
 
+    /// Check whether a group by this name is present.
+    pub fn has_group(&self, name: &str) -> bool {
+        self.group
```

Comment:
This can be shortened and avoid the clone by doing:

```rust
self.group.as_ref().unwrap_or_default().any(|g| g.name == name)
```

Reply:
Doing that returns a reference to the vec, which I don't think can be satisfied by either unwrap_or_default or unwrap_or_else. Also, I think you still need the iter().

Reply:
Alright... came up with a half-way workaround that omits the clone.